api/enterprise-api-ref/reset-password.markdown

```diff
@@ -0,0 +1,106 @@
+---
+layout: default
+title: Reset password API
+published: true
+---
+
+## Request reset password link and token
+
+This call initiates the password reset process by sending a reset password
+link and token to the user's registered email address. User can click the link from the email
+and set a new password in the Mission Portal or invalidate the request.
+Every request has an expiration time equals to 48 hours.
+
+**URI:** https://hub.cfengine.com/api/auth/password/forgot/:username
+
+**Method:** POST
+
+**Example request (curl):**
+
+```console
+curl -X POST \
+  https://hub.cfengine.com/api/auth/password/forgot/admin
+```
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+
+Reset password email successfully sent.
+```
+
+**Responses:**
+
+| HTTP response code       | Description                                                                         |
+|--------------------------|-------------------------------------------------------------------------------------|
+| 200 OK                   | Reset password email successfully sent                                              |
+| 404 Not Found            | User is not found.                                                                  |
+| 422 Unprocessable Entity | Error occurred while email sending. Please contact administrator to reset password. |
+| 429 Too Many Requests    | Time threshold applied on multiple invalid username attempts.                       |
+
+
+## Reset password by token
+
+This call provides possibility to change password by reset password token
+from the [Request reset password link and token][Reset password API#Request reset password link and token] endpoint.
+
+**URI:** https://hub.cfengine.com/api/auth/password/reset/:token
+
+**Method:** POST
+
+**Example request (curl):**
+
+```console
+curl -X POST \
+  --data-raw '{"password": "new password"}' \
+  https://hub.cfengine.com/api/auth/password/reset/v1twknmrLWos
+```
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+
+Reset password email successfully sent.
+```
+
+**Responses:**
+
+| HTTP response code       | Description                                                     |
+|--------------------------|-----------------------------------------------------------------|
+| 200 OK                   | Password successfully changed.                                  |
+| 404 Not Found            | Incorrect password reset token.                                 |
```

Comment:
again here we shouldn't reveal much information so 200 OK or 422 I'd say.

Reply:
unless Mission Portal relies on these return codes? :scream_cat:

Reply:
yes, changed accordingly to api